components/MintButton.js

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react'
+import { Button } from './ui/Buttons'
+import useWeb3 from './hooks/useWeb3'
+import { Body1 } from './ui/Typography'
+
+const MintButton = () => {
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+  const { accounts, contracts } = useWeb3()
+
+  if (!accounts.length) {
+    return <Body1>Please connect to your wallet.</Body1>
+  }
+
+  if (error) {
+    return <Body1>{error}</Body1>
+  }
+
+  const mint = async () => {
+    try {
+      setLoading(true)
+      await contracts.LvlV1Contract.methods.mint().send({
```

Comment:
I like that idea, I was mostly following what people have been doing for Uniswap V2/V3 etc. I think they wanted to make it super explicit in the code since moving versions is meant to be a completely new dApp (for instance, V2 has a dapp and V3 has a dapp)

Reply:
Also, thinking more about this, if we maintain a single dApp portal, we will likely need to interact with V1 and V2 so we can help people bridge to mainnet when the time comes.

Reply:
I like this, is better to be explicit about contract version interaction within the app because transtitioning to a new contract version is not that fun in everything is generic.